SEARCHSPEC.md

```diff
@@ -0,0 +1,1476 @@
+
+
+#### Example: Semantic Data Types in Query Results
+
+When a user issues the following query to the /search endpoint
+
+```
+select
+    ga4gh_type(id, '$ref:https://schemablocks.org/schemas/sb-phenopackets/current/Age.json#properties/age') as id,
+    ga4gh_type(case
+        when blood_type = '' then null
+        else cast(row(
+            case regexp_extract(blood_type, '(\w+)([+-])', 1)
+            when '0' then 'HP:0032442'
+            when 'A' then 'HP:0032370'
+            when 'B' then 'HP:0032440'
+            when 'AB' then 'HP:0032441'
+            else 'error'
+            end,
+            case regexp_extract(blood_type, '(\w+)([+-])', 1)
+            when '0' then 'O'
+            else regexp_extract(blood_type, '(\w+)([+-])', 1)
+            end
+        )
+        as row(id varchar, label varchar))
+    end, 
+'$ref:https://schemablocks.org/schemas/example/blood-group/v1.0.0/BloodGroup.json') as blood_group
+`from search_postgres_pgpc.public.participant
+```
+Then the Search service would respond with:
+
+```
+{
+  "data_model": {
+    "$id": null,
+    "description": "Schema specified by query",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "properties": {
+      "id": {
+        "$ref": "https://schemablocks.org/schemas/sb-phenopackets/current/Age.json#properties/age"
+      },
+      "blood_group": {
+        "$ref": "https://schemablocks.org/schemas/example/blood-group/v1.0.0/BloodGroup.json"
+      }
+    }
+  },
+  "data": [
+    {
+    {
+      "id": "PGPC-44",
+      "blood_group": {"id": "HP:0032442", "label": "O"}
+    }
+  ],
+  "pagination": {
+    "next_page_url": "http://search-api.example.com/search/v1/statement/executing/20200630_221357_00006_4ufsz/ya41766923be262f6ddd468d212c0a1fc9f4797b2/1"
+  }
+}
+```
+
+### SQL Functions (WIP)
+
+The Search API’s SQL dialect has been selected for compatibility with current major open source database platforms including Presto SQL, PostgreSQL, and MySQL, as well as BigQuery. There are occasional name or signature differences, but a Search API implementation atop any of the major database platforms should be able to pass through queries that use the functions listed below with only minor tweaks.
+
+The functions below are a subset of those available in PrestoSQL 341. In a conformant Search API implementation, these functions must behave according to the Presto documentation. To assist with implementations directly on other database platforms, the [PrestoSQL Functions Support Matrix](https://docs.google.com/document/d/1y51qNuoe2ELX9kCOyQbFB4jihiKt2N8Qcd6-zzadIvk) captures the differences between platforms in granular detail. 
+
+*   **Logical Operators**
```

Comment:
It would be better in this whole section to quote the function and operator names individually, eg:

* `AND`, `OR`, `NOT`
* `abs(x)` → [same as input]
* `pi()` → `double`

The asterisks would not go inside the quotes.